AAP-1439: Add section for Access Control Requirements (#412)
RedHatInsights/red-hat-ansible-automation-platform-documentation: downstream/modules/platform/ref-access-control-requirements.adoc

|5432
|TCP
|PostgreSQL
a|Open *only* if the internal database is being used. Otherwise, this port should not be opened

Hybrid mode in a cluster

|27199
|TCP
|Receptor
|ALLOW receptor port across all controllers (for mandatory & automatic control plane peering)
|===

.Hop Nodes
[options="header"]
|===
|Port |Protocol |Service |Required for
|22
|TCP
|SSH
|Installation

|27199
|TCP
|Receptor
a|Mesh

ALLOW connection from controller(s) to receptor port
|===

.Execution Nodes
[options="header"]
|===
|Port |Protocol |Service |Required for
|22
|TCP
|SSH
|Installation

|80
|TCP
|HTTP
|Execution Environment (EE) pulls container images from registry

|443
|TCP
|HTTPS
|Execution Environment (EE) pulls container images from registry

|27199
|TCP
|Receptor
a|Mesh - Nodes directly peered to controllers. No hop nodes involved. 27199 is bidirectional for the execution nodes

ALLOW connections from controller(s) to receptor port (for non-hop connected nodes)

ALLOW connections from hop node(s) to receptor port (if relayed through hop nodes)

|===

.{HubNameStart}
[options="header"]
|===
|Port |Protocol |Service |Required for
|22
|TCP
|SSH
|

|80
|TCP
|HTTP
|
|443
|TCP
|HTTPS

> |

Reviewer: These two rows seem to be missing descriptions for the `Required for` column

Reviewer: I'm sorry @lmalivert I keep doing it every time I see the email. I'll delete my comments. Apologies again for the noise and hope you enjoy the weekend.

> |

Author: No worries @oraNod. Comments coming in while PRs are still WIP throws me off a bit so I'm a stickler for requesting peer-reviews _after_ SME's have provided their tech reviews and I have completed the draft 😸

[id="ref-access-control-requirements_{context}"]

= Access control requirements

[role="_abstract"]

{PlatformName} uses a number of ports to communicate with its services. These ports must be open and available for incoming connection to the {PlatformName} server in order for it to work. Ensure that these ports are available and are not being blocked by a firewall.

The following tables provide the default {PlatformName} destination ports required for each application.

.PostgreSQL
[options="header"]
|===
|Port |Protocol |Service |Required for
|22
|TCP
|SSH
|Remote access during installation
|5432
|TCP
|Postgres
a|Default port

> ALLOW connections from controller(s) to database port

Reviewer 2: ENABLE not allow, throughout.

Reviewer: @ianf77 Is this a preference or is there a style guideline? I'm curious as I would use "allow" when talking about ports. When you open a port you allow traffic through it, whereas enable implies you do something that turns on the ability to send traffic through a port. I feel there is a semantic change in meaning that doesn't really work.

Reviewer: Then again it could just be me and the fact I haven't had coffee yet. :wink:

Reviewer 2: IBM Style guide says : Avoid stating that inanimate objects grant abilities to people, as in "the product allows you to...". Whenever possible, use a
direct, user-focused alternative such as "you can use the product to..." or "with this product, users can..."

Reviewer 2: Editors at a previous company would ruthlessly purge all appearance of allow, so I learned to avoid it.

Reviewer: Thanks for clarifying, Ian. My 2c is that doesn't necessarily apply in this case. In general that's good advice but it seems like it's more in the spirit of "focusing on the task, not the tools".

It's still my view that `s/allow/enable` misses some nuances. When you open a port on the network, you allow traffic through that port. In this scenario the user does not enable any functionality.

I also think the convention with inbound/outbound ports is to "allow traffic" so changing to "enable" just doesn't sound correct to a technical audience.

[id="ref-access-control-requirements_{context}"]

= Access control requirements

[role="_abstract"]

{PlatformName} uses a number of ports to communicate with its services. These ports must be open and available for incoming connection to the {PlatformName} server in order for it to work. Ensure that these ports are available and are not being blocked by a firewall.

The following tables provide the default {PlatformName} destination ports required for each application.

.PostgreSQL
[options="header"]
|===
|Port |Protocol |Service |Required for
|22
|TCP
|SSH
|Remote access during installation
|5432
|TCP
|Postgres
a|Default port

ALLOW connections from controller(s) to database port
|===

.{ControllerNameStart}
[options="header"]
|===
|Port |Protocol |Service |Required for
|22
|TCP
|SSH
|Installation

|80
|TCP
|HTTP

> |UI/API & {ExecEnvShort} pulls

Reviewer 3: What does "execution environment pulls" mean ?

Author: Hi @Spredzy I copied this information, word for word from this [google doc](https://docs.google.com/document/d/1g8oIX5xiZEEvoTmjB1B1REzSCw55aJpNpBSBd4VGv6g/edit#bookmark=id.69cn3hp5zhlf), provided in [AAP-1439](https://issues.redhat.com/browse/AAP-1439) that was created by Phil Griffiths. 

That doc seems to be a collaboration effort from different SMEs. I've had difficulty pinning down an SME who contributed to that doc to tell me what some of those comments mean. Should I remove "execution environment pulls" from the table?

Author: @Spredzy I've received an explanation of "EE pulls." =  "Pulls execution environments from your chosen container registry" and made the edit.

|443
|TCP
|HTTPS
|UI/API & {ExecEnvShort} pulls

|5432
|TCP
|PostgreSQL
a|Open *only* if the internal database is being used. Otherwise, this port should not be opened

Hybrid mode in a cluster

|27199
|TCP
|Receptor
|ALLOW receptor port across all controllers (for mandatory & automatic control plane peering)
|===

.Hop Nodes
[options="header"]
|===
|Port |Protocol |Service |Required for
|22
|TCP
|SSH
|Installation

|27199
|TCP
|Receptor
a|Mesh

ALLOW connection from controller(s) to receptor port
|===

.Execution Nodes
[options="header"]
|===
|Port |Protocol |Service |Required for
|22
|TCP
|SSH
|Installation

> |80

Reviewer 3: Execution nodes don't need to have port 80 and 443 open.

|TCP
|HTTP
|Execution Environment (EE) pulls container images from registry

> |443

Reviewer 3: ditto

Author: @Spredzy Thanks, I'll remove ports 80 and 443 from Execution Nodes.

|TCP
|HTTPS
|Execution Environment (EE) pulls container images from registry

|27199
|TCP
|Receptor
a|Mesh - Nodes directly peered to controllers. No hop nodes involved. 27199 is bidirectional for the execution nodes

ALLOW connections from controller(s) to receptor port (for non-hop connected nodes)

ALLOW connections from hop node(s) to receptor port (if relayed through hop nodes)

|===

.{HubNameStart}
[options="header"]
|===
|Port |Protocol |Service |Required for
|22
|TCP
|SSH
|Installation

|80
|TCP
|HTTP
|User interface
|443
|TCP
|HTTPS
|User interface
|5432
|TCP
|PostgreSQL
a|Open *only* if the internal database is being used. Otherwise, this port should not be opened
|===

> .Service Catalog

Reviewer 3: IIRC It's Services Catalog (an s at the end of Service)

[options="header"]
|===
|Port |Protocol |Service |Required for
|22
|TCP
|SSH
|Installation
|443
|TCP
|HTTPS
|Access to Service Catalog user interface
|5432
|TCP
|PostgreSQL
a|Open *only* if the internal database is being used. Otherwise, this port should not be opened
|===

.{InsightsName}
[options="header"]
|===
|URL |Required for
|link:http://api.access.redhat.com:443[http://api.access.redhat.com:443] |General account services, subscriptions
|link:https://cert-api.access.redhat.com:443[https://cert-api.access.redhat.com:443] |Insights data upload
|link:https://cert.cloud.redhat.com:443[https://cert.cloud.redhat.com:443] |Inventory upload and Cloud Connector connection
|link:https://cloud.redhat.com[https://cloud.redhat.com] |Access to Insights dashboard
|===

.Automation Hub
[options="header"]
|===
|URL |Required for

> |link:https://console.redhat.com:443[https://console.redhat.com:443] |General account services, subscriptions

Reviewer 4: Do the URLs for specific ports need to be "live" links from the docs?

[id="ref-access-control-requirements_{context}"]

= Access control requirements

[role="_abstract"]

> {PlatformName} uses a number of ports to communicate with its services. These ports must be open and available for incoming connection to the {PlatformName} server in order for it to work. Ensure that these ports are available and are not being blocked by a firewall.

Reviewer 5: ```suggestion
{PlatformName} uses a number of ports to communicate with its services. Ensure that these ports are open and available in the firewall settings to allow for incoming connection to the {PlatformName} server.
```

Reviewer 5: Just a suggestion. There may be a better way to phrase this, but I wanted to try to consolidate the sentences.
